app/main/messaging.py

```diff
@@ -0,0 +1,392 @@
+                        is_public=True)
+        db.session.add(agency)
+        db.session.commit()
+
+    new_incident = IncidentReport(
+        agency=agency,
+        vehicle_id=vehicle_id,
+        license_plate=license_plate if license_plate else None,
+        duration=timedelta(minutes=duration),
+        description=description,
+        location=Location(
+            latitude=lat,
+            longitude=lon,
+            original_user_text=location
+        ),
+        picture_url=picture_url if picture_url else None,
+        user=User.query.filter_by(phone_number=phone_number).first()
+    )
+    db.session.add(new_incident)
+    db.session.commit()
+
+    return new_incident
+
+
+def handle_start_report(step, twiml):
+    step = 1
+    twiml.message('What is your location? Be specific! (e.g. "34th and '
+                  'Spruce in Philadelphia PA")')
+    return step
+
+
+def handle_location_step(body, step, twiml):
+    validator_form = IncidentReportForm()
+    errors = data_errors(form=validator_form, field=validator_form.location,
+                         data=body)
+
+    if geocode(body) == (None, None):
+        errors.append('We could not find that location. Please respond with a '
+                      'full address including city and state.')
+    if len(errors) == 0:
+        location = body
+        step += 1
+        agencies = Agency.query.filter_by(is_official=True).order_by(
+            Agency.name).all()
+        letters = all_strings(len(agencies) + 1)  # one extra letter for Other
+        agencies_listed = get_agencies_listed(agencies, letters)
+        twiml.message('Which agency operates the vehicle you see idling? '
+                      'Select from the following list or enter {} for Other.'
+                      .format(letters[-1]))
+        twiml.message(agencies_listed)
+    else:
+        location = ''
+        reply_with_errors(errors, twiml, 'location')
+
+    return location, step
+
+
+def handle_agency_step(body_upper, step, twiml):
+    body_upper = body_upper.upper()
+    agencies = Agency.query.filter_by(is_official=True).all()
+    letters = all_strings(len(agencies) + 1)  # one extra letter for Other
+    letters_to_agency = dict(zip(letters, agencies))
+
+    if body_upper == letters[-1]:  # Other
+        step += 1
+        agency_name = ''
+        twiml.message('You selected Other. What is the name of the agency '
+                      'which operates the vehicle?')
+
+    elif body_upper in letters_to_agency.keys():
+        step += 2
+        agency_name = letters_to_agency[body_upper].name
+        twiml.message('What is the license plate number? Reply "no" to skip. '
+                      '(e.g. MG1234E)')
+
+    else:
+        agency_name = ''
+        twiml.message('Please enter a letter A through {}.'
+                      .format(letters[-1]))
+
+    return agency_name, step
+
+
+def handle_other_agency_step(body, step, twiml):
```

Comment:
LOL can we rename this function? like `handle_license_plate_step`. suggestion: i might leave out the `_step` off the function variable names, e.g. `handle_agency(...)`